hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -548,7 +553,9 @@ public void executeOnEntriesInternal(EntryProcessor entryProcessor, Predicate pr
         } finally {
             for (int i = 0; i < resultingKeyValuePairs.size(); i += 2) {
                 Data key = resultingKeyValuePairs.get(i);
-                invalidateNearCache(serializeKeys ? key : toObject(key));
+                if (!(entryProcessor instanceof ReadOnly)) {
```

Comment:
similar to above, we don't need to do anything here, if we won't invalidate the near cache

Reply:
Updated same as above